File: src/LetterRepositoryUI/src/app/common/models/visit.ts
export class Visit {
    ObId = '';
    VisitorId = 0;
    Description = '';
    VisitDate = new Date();
    DepartmentId = 0;
    Purpose = 0;
    SeqId = 0;
    VisitorToken = 'AUTO GENERATED';
    IsWorkDone = false;
    Progress = this.IsWorkDone ? 'Finished' : 'Pending';
    Department = "";
}

File: src/LetterRepository.api/Models/Visit.cs
using System;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace LetterRepository.api.Models
{
    public class Visit
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string ObId { get; set; }
        public long VisitorId { get; set; }
        public long Purpose { get; set; }
        public string Description { get; set; }
        public long DepartmentId { get; set; }
        [BsonIgnore]
        public string Department { get; set; }

        [BsonDateTimeOptions(Kind = DateTimeKind.Local)]
        public DateTime VisitDate { get; set; }
        public int SeqId { get; set; }
        public string VisitorToken { get; set; }
        public bool IsWorkDone { get; set; }
        public string Progress { get; set; }
    }
}